### PR TITLE
Cd lint prettier optimization

### PR DESCRIPTION
**Pull request by fredbonin**
I modified the way eslint and typecheck is done so that it is done at the project level instead of individual projects as suggested here : https://typescript-eslint.io/docs/getting-started/linting/monorepo/.

This is necessary because of a known bug when there are multiple modules : https://github.com/typescript-eslint/typescript-eslint/issues/1192

This will speed up internal ci/cd pipeline from 45m to ~4m

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by Brian969** [comment]
Love the progress on this and great teamwork Fred and Antoine.
- Excluding the UI (alpha code) at this time is acceptable, but it needs to be brought in scope when it does get worked on/before it becomes prod
- excluding src/deployments/cdk/tools.ts, src/deployments/cdk/toolkit.ts, src/deployments/cdk/cdk.ts is not acceptable - we need to spend the time to make the code fixes to bring these in scope of typechecking
- likely same applies to src/core/runtime/test/*

**Comment by fredbonin** [comment]
I definitely agree that we need to fully lint and typecheck the whole project but would you like to do it as part as this PR ? The project did not check the files above before this PR. 

We can create another issue with another PR for these files specifically so that we can benefit for the shorter time now.

**Comment by antoineaws** [comment]
We are not far off from what Brian is suggesting. Note the remaining items below (~5 files):
- Typecheck will run against all sub-projects except below:
  - src/ui 
  - [ ] (Remaining) src/deployments/cdk/tools.ts (excluded currently, but can be fixed)
  - [ ] (Remaining) src/deployments/cdk/toolkit.ts (excluded currently, but we can fix, might need to check with Dustin on line 88)
  - [ ] (Remaining) src/deployments/cdk/cdk.ts (excluded currently, but can be fixed)
  - [ ] (Remaining) src/core/runtime/test/* (total 2 files)
- Lint will run against all sub-projects except below:
  - src/ui (project currently isn't linting)

As for ui project, linting can be enabled in another PR as Brian mentioned, it is alpha currently.
 
I can help out clear up the remaining typecheck fixes?

**Comment by Brian969** [comment]
@fredbonin - I understand effort to fix the above files is likely under 1hr, if so, I'd love included in this PR.  If more, yes, let move to a new PR.

**Comment by fredbonin** [comment]
All done !

File: src/lib/docs-gen/src/helpers/translation-extract-helper.ts [hunk 1]
```diff
@@ -39,26 +41,32 @@ export class TranslationExtractHelper {
...
         ($1, $2, $3) => {
           return `${$2.toUpperCase() + $3.toLowerCase()}`;
         },
       );
     return newTitle;
   }
   private translate(key: string, selectedLanguage: string) {
-    if (selectedLanguage === 'fr') return frenchDict.fr[key];
-    else return englishDict.en[key];
+    if (selectedLanguage === 'fr') {
```
Comment (hickeydh-aws) [inline src/lib/docs-gen/src/helpers/translation-extract-helper.ts:59]: The else here is probably extraneous
`    if (selectedLanguage === 'fr') {
      return frenchDict.fr[key];
    }
    return englishDict.en[key];
      `

**Review by hickeydh-aws — APPROVED** [review]

**Pull request merged**